src/main/java/redis/clients/jedis/PipelineBase.java

```diff
@@ -1975,15 +1975,13 @@
     return getResponse(BuilderFactory.BYTE_ARRAY_LIST);            
   }
 
-  public Response<Object> sendCommand(ProtocolCommand cmd, String... args){
-    String key = args.length > 0 ? args[0] : cmd.toString();
-    getClient(key).sendCommand(cmd, args);
+  public Response<Object> sendCommand(final String sampleKey, final ProtocolCommand cmd, final String... args) {
```

Comment:
@sazzad16 I think we should just deprecate these methods, once you have them in the MultiKeyPipelineBase you don't need these.
I don't think we need those for ShardedJedisPipeline or just move them there with the sample keys.